overlord/snapstate/handlers_components.go

```diff
@@ -561,7 +561,7 @@ func (m *SnapManager) doSetupKernelModules(t *state.Task, finalStatus state.Stat
 	}
 
 	// kernel-modules components already in the system
-	kmodComps := snapSt.Sequence.ComponentsWithTypeForRev(snapsup.Revision(), snap.KernelModulesComponent)
+	kmodComps := snapSt.Sequence.ComponentsWithTypeForRev(snapSt.Current, snap.KernelModulesComponent)
```

Comment:
can you explain this change?

Reply:
From my understanding, this call to `snapSt.Sequence.ComponentsWithTypeForRev` was supposed to gather all kernel module components for the currently installed revision of the snap. This was correct until now, as `snapsup.Revision()` was equal to the current revision of the snap. Now snapsup will contain the revision of the new snap, while snapSt.Current will be the current revision.

I could be understanding this code wrong, though. @alfonsosanchezbeato should probably check my work here.

Reply:
I think that the change is correct, thanks

Reply:
This change wasn't quite right, I made some changes in d244566436b2cb4f382353bce5b36fe2463d906a.